src/inference/AdvancedSMC.jl

```diff
@@ -281,17 +281,21 @@ function AbstractMCMC.step(
 )
     # Reset the VarInfo before new sweep.
     reset_num_produce!(vi)
-    set_retained_vns_del_by_spl!(vi, spl)
     resetlogp!(vi)
 
+    # Create reference particle for which the samples will be retained.
+    reference = AdvancedPS.forkr(AdvancedPS.Trace(model, spl, vi))
```

Comment:
nice catch - maybe add a comment referencing the Github issue so that we won't incidentally re-introduce the same bug? 

EDIT: I see there is a comment in the tests.